src/AdoPat/PatClient.cs

```diff
@@ -0,0 +1,139 @@
+
+        private ITokensHttpClientWrapper client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PatClient"/> class.
+        /// </summary>
+        /// <param name="client">Any class which implements the <see cref="ITokensHttpClientWrapper"/> interface.</param>
+        public PatClient(ITokensHttpClientWrapper client)
+        {
+            this.client = client;
+        }
+
+        /// <summary>
+        /// Creates a new PAT.
+        /// </summary>
+        /// <param name="patTokenCreateRequest">
+        /// XXX: Not documented by Visual Studio client libraries. Should we depend on this or
+        /// should this method use the corresponding primitives as arguments? Not yet clear.
+        /// </param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A <see cref="Task"/> returning a <see cref="PatTokenResult"/>.</returns>
+        public async Task<PatToken> CreatePatAsync(
+            PatTokenCreateRequest patTokenCreateRequest,
+            CancellationToken cancellationToken = default)
+        {
+            var patTokenResult = await this.client.CreatePatAsync(
+                patTokenCreateRequest,
+                cancellationToken: cancellationToken)
+            .ConfigureAwait(false);
+
+            if (patTokenResult.PatToken == null)
+            {
+                throw new PatClientException($"Failed to create PAT during regeneration: {patTokenResult.PatTokenError}");
+            }
+
+            return patTokenResult.PatToken;
+        }
+
+        /// <summary>
+        /// Gets all active PATs.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>The set of active PATs.</returns>
+        public async Task<ISet<PatToken>> GetActivePatsAsync(CancellationToken cancellationToken = default)
+        {
+            // Initialize a PagedPatTokens so that we can use the continuation token
+            // in the scope of the conditional of the do while loop below. Azure
+            // DevOps will return the empty string when there are no more pages.
+            var pagedPatTokens = new PagedPatTokens(continuationToken: string.Empty, patTokens: new List<PatToken>());
+            var pats = new HashSet<PatToken>();
+            do
+            {
+                pagedPatTokens = await this.client.ListPatsAsync(
+                    displayFilterOption: DisplayFilterOptions.Active,
+                    top: PageSize,
+                    continuationToken: pagedPatTokens.ContinuationToken,
+                    sortByOption: SortByOptions.DisplayDate,
+                    cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+
+                foreach (PatToken pat in pagedPatTokens.PatTokens)

```

Comment:
We should add a null check here if pagedPatTokens can be null.

Reply:
I really hope that's not the case, but I'll check their docs again to be sure. I feel like it would be silly if it could be `null` and they could just provide an empty list instead...

Reply:
Oh! It's not possible because **we** create the `pagedPatTokens` above with
```csharp
var pagedPatTokens = new PagedPatTokens(continuationToken: string.Empty, patTokens: new List<PatToken>());
```
I don't think there's a need for a `null` check here, @shalinikhare27.

Reply:
Hmm... just kidding. It's still _technically_ possible because we reassign `pagedPatTokens`.

Reply:
OK. I went spelunking **waaay** deep into the implementation details of the Visual Studio client code we're using here and found that the underlying methods are generic and do
```csharp
return default(T);
```
I don't know enough about C# yet, but I don't _think_ that usually returns `null`. Do either of you know, @shalinikhare27 or @kyle-rader?

Reply:
Ugh. That's **also** not entirely correct. `default(T)` definitely could be `null` depending on the type, but there are also earlier returns in that same function that make use of `T`.

Reply:
@Haard30 and I dove pretty deep into the code for the library we're using here and it seems pretty unlikely that we'll get a `null` here without getting some other kind of exception first, so I'm inclined to not do the `null` checking here. Opinions, @shalinikhare27 &amp; @kyle-rader?

Reply:
@shalinikhare27 and I talked through this as well and I think we're in agreement that it's much more likely that some exception will be raised before we ever get a `null` for either `pagedPatTokens` or `pagedPatTokens.PatTokens`.

In any event, even if we did get a `null`, that's not exactly a recoverable situation and it means something else has gone wrong. My plan is to handle exceptions further up the stack in `PatManager`, so I think it's OK to let a `NullReferenceException` bubble up to there. Would still like to get your thoughts, @kyle-rader.